Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.cpp

```diff
@@ -366,9 +367,17 @@ void LayerTreeHost::deviceOrPageScaleFactorChanged()
     m_compositor->setScaleFactor(m_webPage.deviceScaleFactor() * m_viewportController.pageScaleFactor());
 }
 
+PlatformDisplayID LayerTreeHost::displayID() const
+{
+    return m_compositor->displayRefreshMonitor().displayID();
+}
+
 RefPtr<DisplayRefreshMonitor> LayerTreeHost::createDisplayRefreshMonitor(PlatformDisplayID displayID)
 {
-    return m_compositor->displayRefreshMonitor(displayID);
+    auto& displayRefreshMonitor = m_compositor->displayRefreshMonitor();
+    if (displayRefreshMonitor.displayID() == displayID)
+        return Ref { displayRefreshMonitor };
```

Comment:
Can a mismatch here happen?

Reply:
I don't think it can happen now, so I'll add an assert instead.